packages/mco/hooks/disaster-recovery.ts

```diff
@@ -0,0 +1,180 @@
+import * as React from 'react';
+import { useDeepCompareMemoize } from '@odf/shared/hooks/deep-compare-memoize';
+import { getName, getNamespace } from '@odf/shared/selectors';
+import { referenceForModel } from '@odf/shared/utils';
+import {
+  useK8sWatchResources,
+  ObjectReference,
+} from '@openshift-console/dynamic-plugin-sdk';
+import {
+  DRPolicyModel,
+  DRClusterModel,
+  DRPlacementControlModel,
+  ACMManagedClusterModel,
+} from '../models';
+import {
+  DRPolicyKind,
+  DRClusterKind,
+  DRPlacementControlKind,
+  CustomDRPlacementControlKind,
+  PeerStatus,
+  DRClusterInfoType,
+  DRClusterStatus,
+  ACMManagedClusterKind,
+} from '../types';
+import {
+  filerDRClustersUsingDRPolicy,
+  filterDRPolicyUsingDRPolicyRef,
+  findDRPolicyRefFromDRPlacementControl,
+  isPeerReadyAndAvailable,
+  getManagedClusterAvailableCondition,
+  filerManagedClusterUsingDRCluster,
+} from '../utils';
+
+const createDRClusterInfo = (
+  drClusters: DRClusterKind[],
+  managedClusters: ACMManagedClusterKind[]
+): DRClusterInfoType[] =>
+  drClusters?.map((drCluster) => {
+    const clusterName = getName(drCluster);
+    const usedManagedCluster = filerManagedClusterUsingDRCluster(
+      drCluster,
+      managedClusters
+    );
+    const condition = getManagedClusterAvailableCondition(usedManagedCluster);
+    return {
+      clusterName: clusterName,
+      clusterNamespace: getNamespace(drCluster),
+      region: drCluster?.spec?.region,
+      status: drCluster?.status?.phase as DRClusterStatus,
+      isClusterAvailable: !!condition,
+      lastAvailableTime: condition?.lastTransitionTime,
+    };
+  });
+
+const resources = (name?: string, namespace?: string) => ({
+  drPlacementControls: {
+    name: name,
+    namespace: namespace,
+    kind: referenceForModel(DRPlacementControlModel),
+    namespaced: true,
+    isList: true,
+  },
+  drPolicies: {
+    kind: referenceForModel(DRPolicyModel),
+    namespaced: false,
+    isList: true,
+  },
+  drClusters: {
+    kind: referenceForModel(DRClusterModel),
+    namespaced: false,
+    isList: true,
+  },
+  managedClusters: {
+    kind: referenceForModel(ACMManagedClusterModel),
+    isList: true,
+    namespaced: false,
+  },
+});
+
+export const useWatchDRPlacementControlResources: UseK8sWatchResource = (
```

Comment:
nits:
just to shorten the name
```suggestion
export const useWatchDRPCResources: UseK8sWatchResource = (
```